### PR TITLE
Release v2.0.0

### PR DESCRIPTION
**Pull request by oed**
This release branch is up for review.

### CODE REVIEW

File: contracts/Proxy.sol [hunk 1]
```diff
@@ -1,45 +1,15 @@
...
 
-//   function () payable {}
 
-//   function forward(address destination, uint value, bytes data) onlyOwner {
-//     if (!destination.call.value(value)(data)) { throw; }
-//     Forwarded(destination, value, data);
-//   }
-// }
+contract Proxy is Controlled {
+    event LogForwarded (address indexed destination, uint value, bytes data);
```
Comment (pelle) [inline contracts/Proxy.sol]: @oed was it necessary to rename these? There may be code depending on these names. Did it come from the security review?

File: contracts/MetaIdentityManager.sol [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+    /// @param owner Key who can use this contract to control proxy. Given full power
+    /// @param recoveryKey Key of recovery network or address from seed to recovery proxy
+    /// @param destination Address of contract to be called after proxy is created
+    /// @param data of function to be called at the destination contract
+    function createIdentityWithCall(address owner, address recoveryKey, address destination, bytes data) public validAddress(recoveryKey) {
+        Proxy identity = new Proxy();
+        owners[identity][owner] = now - adminTimeLock; // This is to ensure original owner has full power from day one
+        recoveryKeys[identity] = recoveryKey;
+        LogIdentityCreated(identity, msg.sender, owner,  recoveryKey);
+        identity.forward(destination, 0, data);
```
Comment (pelle) [inline contracts/MetaIdentityManager.sol]: Should it check for the presence of the destination before calling it?

File: contracts/TxRelay.sol [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    // Note: This is a local nonce.
+    // Different from the nonce defined w/in protocol.
+    mapping(address => uint) nonce;
+
+    // This mapping specifies a whitelist of allowed senders for transactions.
+    // There can be one whitelist per ethereum account, which is the owner of that
+    // whitelist. Users can specify which whitelist they want to use when signing
+    // a transaction. They can use their own whitelist, a whitelist belonging
+    // to another account, or skip using a whitelist by specifying the zero address.
+    mapping(address => mapping(address => bool)) public whitelist;
```
Comment (pelle) [inline contracts/TxRelay.sol]: I thought we had decided to go with a simple, single mapping here. But I guess it's fine. It's a nice simple elegant implementation.

File: contracts/MetaIdentityManager.sol (hunk 2 above)
Comment (oed) [reply contracts/MetaIdentityManager.sol]: Why? Calling the zero address will have the same effect as calling any other random address that isn't used.

**Pull request merged**